cpp/include/raft/neighbors/cagra.cuh

```diff
@@ -48,12 +57,13 @@ namespace raft::neighbors::cagra {
  *
  * The following distance metrics are supported:
  * - L2Expanded
+ * - InnerProduct
  *
  * Usage example:
  * @code{.cpp}
  *   using namespace raft::neighbors;
  *   // use default index parameters
- *   ivf_pq::index_params build_params;
+ *   ivf_pq::index_params = cagra::get_default_ivf_pq_build_params(dataset);
```

Comment:
Can't add constructors to the `ivf_pq::index_params` struct because the aggregate assertion [here](https://github.com/rapidsai/raft/blob/317a61cad71b5f83424d2481eb23b1d1b5817f40/cpp/include/raft/neighbors/ivf_pq_types.hpp#L148). A helper is needed. One thing that can be done is that instead of a constructor within the `index_params` struct, we can have an helper `get_default_index_params` outside of the struct, but in `ivf_pq_types.hpp`

Reply:
Honestly, I'd just remove the static assertion there. It's not necessary.

Reply:
Here we are constructing _CAGRA specific_ `ivf_pq::index_params`. I do not thing the constructor of `ivf_pq::index_params` shall have the job to define parameters according to CAGRA's requirements.

For consistent experience, the user just need to fill the `metric` field of `cagra::index_params`, which is inherited from [ann::index_params](https://github.com/rapidsai/raft/blob/71a19a2495b74ef726d7d95a0c953e16b3c86d67/cpp/include/raft/neighbors/ann_types.hpp#L34), and call `cagra::build` as usual.

Users of `build_knn_graph` are advanced users who are using `build_knn_graph` & `optimize` in a separate step. Other algos don't have these steps, this is custom for CAGRA. Before this PR, the user either passes a manually filled `ivf_pq::index_params`, or just uses a default. I think a helper function is an improvement.